### PR TITLE
Fix using Bazel-built libraries by cmake_external and configure_make

### PR DESCRIPTION
**Pull request by irengrig**
Please see the description of changes in commit messages.

### CODE REVIEW

File: tools/build_defs/cmake.bzl [hunk 1]
```diff
@@ -77,6 +77,7 @@ def _create_configure_script(configureParameters):
         dict(ctx.attr.cache_entries),
         dict(ctx.attr.env_vars),
         ctx.attr.cmake_options,
+        inputs.include_dirs,
```
Comment (aehlig) [inline tools/build_defs/cmake.bzl]: Don't pass keyword arguments positionally (this is also wrong for the next argument, but at least do not make it worse).

Independently of this change, the limit for the number of positional arguments that is still managable seems to be exceeded for this function.

File: tools/build_defs/shell_toolchain/toolchains/impl/default_commands.bzl [hunk 2]
```diff
@@ -73,10 +73,12 @@ if [[ -f $1 ]]; then
   return 0
 fi
 
-local children=$(find $1 -maxdepth 1 -mindepth 1)
-for child in $children; do
-  ##symlink_to_dir## $child $target
-done
+if [[ -d $1 || -L $1 ]]; then
```
Comment (aehlig) [inline tools/build_defs/shell_toolchain/toolchains/impl/default_commands.bzl]: `[[` is a `bash` extension; as there are already `bash`-only constructs in the `text`, I assume it is OK, but I couldn't find the documentation about whether this is called by a generic shell (like `sh`) or by `bash` specifically (the directory name `shell_toolchain` seems to hint at the former).

Comment (irengrig) [reply tools/build_defs/shell_toolchain/toolchains/impl/default_commands.bzl]: Thank you for the remark; bash is assumed, will clarify that.

**Comment by irengrig** [comment]
Thank you for the review!

**Pull request merged**